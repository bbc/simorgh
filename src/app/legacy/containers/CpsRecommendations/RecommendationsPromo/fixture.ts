export const cpsRecommendation = {
  headlines: {
    shortHeadline: 'Meet boys who dey convert cassava to electricity',
    headline: 'Meet boys who dey convert cassava to electricity',
  },
  locators: {
    assetUri: '/pidgin/44508901',
    cpsUrn: 'urn:bbc:content:assetUri:/pidgin/44508901',
  },
  summary:
    "James den Kwesi Ansah want convert most of Ghana's cassava waste into cheap electricity supply for schools, hospitals den villages.",
  timestamp: 1529397290,
  cpsType: 'MAP',
  indexImage: {
    id: '102105505',
    subType: 'index',
    href: 'http://c.files.bbci.co.uk/C576/production/_102105505_p06bhwsd.jpg',
    path: '/cpsprodpb/C576/production/_102105505_p06bhwsd.jpg',
    height: 549,
    width: 976,
    altText: 'Kwesi and James dey check radio',
    caption:
      'James Ansah den Kwesi Ansah dey experiment plus dema electricity inovation from cassava',
    copyrightHolder: 'BBC',
    type: 'image',
  },
  id: 'urn:bbc:ares::asset:pidgin/44508901',
  type: 'cps',
};

export const optimoRecommendation = {
  locators: {
    optimoUrn: 'urn:bbc:optimo:asset:crg7rvwrxdlo',
    canonicalUrl: 'https://www.bbc.com/turkce/articles/crg7rvwrxdlo',
  },
  timestamp: 1660891088144,
  suitableForSyndication: true,
  language: 'tr',
  headlines: {
    seoHeadline:
      'Merkez Bankası faiz kararı: Politika faizi neden indirildi, enflasyonu düşürmek için ne yapmalı?',
    promoHeadline: {
      blocks: [
        {
          type: 'text',
          model: {
            blocks: [
              {
                type: 'paragraph',
                model: {
                  text: 'Merkez Bankası politika faizini neden indirdi, enflasyonu düşürmek için ne yapmalı?',
                  blocks: [
                    {
                      type: 'fragment',
                      model: {
                        text: 'Merkez Bankası politika faizini neden indirdi, enflasyonu düşürmek için ne yapmalı?',
                        attributes: [],
                      },
                    },
                  ],
                },
              },
            ],
          },
        },
      ],
    },
  },
  images: {
    defaultPromoImage: {
      blocks: [
        {
          type: 'altText',
          model: {
            blocks: [
              {
                type: 'text',
                model: {
                  blocks: [
                    {
                      type: 'paragraph',
                      model: {
                        text: 'dolar TL ',
                        blocks: [
                          {
                            type: 'fragment',
                            model: { text: 'dolar TL ', attributes: [] },
                          },
                        ],
                      },
                    },
                  ],
                },
              },
            ],
          },
        },
        {
          type: 'rawImage',
          model: {
            width: 1023,
            height: 575,
            locator: '98dd/live/59717db0-1f53-11ed-aa9d-57accb179502.jpg',
            originCode: 'cpsprodpb',
            copyrightHolder: 'Getty Images',
            suitableForSyndication: true,
          },
        },
      ],
    },
  },
  summary: {
    blocks: [
      {
        type: 'text',
        model: {
          blocks: [
            {
              type: 'paragraph',
              model: {
                text: "Kredi faizleri artıp TL, Dolar karşısında değer kaybetmeyi sürdürürken Merkez Bankası dün politika faizini 100 baz puanlık indirimle yüzde 13'e düşürdü. Peki TCMB faizi neden düşürüyor? İzlenilen 'enflasyonla büyüme' politikasının Türkiye'ye faturası ne oluyor? Prof. Dr. Selva Demiralp'in analizi",
                blocks: [
                  {
                    type: 'fragment',
                    model: {
                      text: "Kredi faizleri artıp TL, Dolar karşısında değer kaybetmeyi sürdürürken Merkez Bankası dün politika faizini 100 baz puanlık indirimle yüzde 13'e düşürdü. Peki TCMB faizi neden düşürüyor? İzlenilen 'enflasyonla büyüme' politikasının Türkiye'ye faturası ne oluyor? Prof. Dr. Selva Demiralp'in analizi",
                      attributes: [],
                    },
                  },
                ],
              },
            },
          ],
        },
      },
    ],
  },
  passport: {
    language: 'tr',
    home: 'http://www.bbc.co.uk/ontologies/passport/home/Turkce',
    locator: 'urn:bbc:optimo:asset:crg7rvwrxdlo',
    availability: 'AVAILABLE',
    taggings: [
      {
        predicate: 'http://www.bbc.co.uk/ontologies/bbc/assetType',
        value:
          'http://www.bbc.co.uk/things/22ea958e-2004-4f34-80a7-bf5acad52f6f#id',
      },
      {
        predicate: 'http://www.bbc.co.uk/ontologies/creativework/about',
        value:
          'http://www.bbc.co.uk/things/06381136-9dc0-430f-bf48-b20106f0b18f#id',
      },
      {
        predicate: 'http://www.bbc.co.uk/ontologies/creativework/about',
        value:
          'http://www.bbc.co.uk/things/066edbdf-d295-42ce-9df2-caf00eefb307#id',
      },
      {
        predicate: 'http://www.bbc.co.uk/ontologies/audience/motivation',
        value:
          'http://www.bbc.co.uk/things/b1a660eb-ef14-4645-b4cd-d7bd939ce443#id',
      },
      {
        predicate: 'http://www.bbc.co.uk/ontologies/creativework/about',
        value:
          'http://www.bbc.co.uk/things/ca170ae3-99c1-48db-9b67-2866f85e7342#id',
      },
      {
        predicate: 'http://www.bbc.co.uk/ontologies/audience/motivation',
        value:
          'http://www.bbc.co.uk/things/7047e74c-b9ae-4c02-a4a8-748df451ac58#id',
      },
      {
        predicate: 'http://www.bbc.co.uk/ontologies/creativework/format',
        value:
          'http://www.bbc.co.uk/things/46c0517d-9927-4d1a-9954-8c63a3f7a888#id',
      },
      {
        predicate: 'http://www.bbc.co.uk/ontologies/creativework/about',
        value:
          'http://www.bbc.co.uk/things/a30418d5-3efe-415d-b2d2-1480eaf2eb4a#id',
      },
      {
        predicate: 'http://www.bbc.co.uk/ontologies/bbc/infoClass',
        value:
          'http://www.bbc.co.uk/things/0db2b959-cbf8-4661-965f-050974a69bb5#id',
      },
      {
        predicate: 'http://www.bbc.co.uk/ontologies/audience/motivation',
        value:
          'http://www.bbc.co.uk/things/bf928ac3-b3bd-4d47-924e-cca1bdc29174#id',
      },
    ],
    schemaVersion: '1.4.0',
    publishedState: 'PUBLISHED',
    predicates: {
      assetType: [
        {
          value:
            'http://www.bbc.co.uk/things/22ea958e-2004-4f34-80a7-bf5acad52f6f#id',
          type: 'assetType',
        },
      ],
      motivation: [],
      infoClass: [
        {
          value:
            'http://www.bbc.co.uk/things/0db2b959-cbf8-4661-965f-050974a69bb5#id',
          type: 'infoClass',
        },
      ],
      formats: [
        {
          value:
            'http://www.bbc.co.uk/things/46c0517d-9927-4d1a-9954-8c63a3f7a888#id',
          thingLabel: 'News report',
          thingUri:
            'http://www.bbc.co.uk/things/46c0517d-9927-4d1a-9954-8c63a3f7a888#id',
          thingId: '46c0517d-9927-4d1a-9954-8c63a3f7a888',
          thingType: ['tagging:TagConcept', 'tagging:Format'],
          thingSameAs: [],
          thingEnglishLabel: 'Report',
          thingPreferredLabel: 'Report',
          thingLabelLanguage: 'tr',
          type: 'formats',
        },
      ],
      about: [
        {
          value:
            'http://www.bbc.co.uk/things/06381136-9dc0-430f-bf48-b20106f0b18f#id',
          thingLabel: 'Lira',
          thingUri:
            'http://www.bbc.co.uk/things/06381136-9dc0-430f-bf48-b20106f0b18f#id',
          thingId: '06381136-9dc0-430f-bf48-b20106f0b18f',
          thingType: ['tagging:TagConcept', 'core:Theme', 'core:Thing'],
          thingSameAs: [
            'http://www.wikidata.org/entity/Q172872',
            'http://dbpedia.org/resource/Turkish_lira',
          ],
          thingEnglishLabel: 'Turkish Lira (TRY)',
          type: 'about',
        },
        {
          value:
            'http://www.bbc.co.uk/things/066edbdf-d295-42ce-9df2-caf00eefb307#id',
          thingLabel: 'Türkiye ekonomisi',
          thingUri:
            'http://www.bbc.co.uk/things/066edbdf-d295-42ce-9df2-caf00eefb307#id',
          thingId: '066edbdf-d295-42ce-9df2-caf00eefb307',
          thingType: ['core:Theme', 'core:Thing', 'tagging:TagConcept'],
          thingSameAs: [
            'http://www.wikidata.org/entity/Q431095',
            'http://dbpedia.org/resource/Economy_of_Turkey',
          ],
          thingEnglishLabel: 'Economy of Turkey',
          type: 'about',
        },
        {
          value:
            'http://www.bbc.co.uk/things/a30418d5-3efe-415d-b2d2-1480eaf2eb4a#id',
          thingLabel: 'Türkiye',
          thingUri:
            'http://www.bbc.co.uk/things/a30418d5-3efe-415d-b2d2-1480eaf2eb4a#id',
          thingId: 'a30418d5-3efe-415d-b2d2-1480eaf2eb4a',
          thingType: ['tagging:TagConcept', 'core:Place', 'core:Thing'],
          thingSameAs: ['http://sws.geonames.org/298795/'],
          thingEnglishLabel: 'Turkey',
          type: 'about',
        },
        {
          value:
            'http://www.bbc.co.uk/things/ca170ae3-99c1-48db-9b67-2866f85e7342#id',
          thingLabel: 'Ekonomi',
          thingUri:
            'http://www.bbc.co.uk/things/ca170ae3-99c1-48db-9b67-2866f85e7342#id',
          thingId: 'ca170ae3-99c1-48db-9b67-2866f85e7342',
          thingType: ['core:Theme', 'core:Thing', 'tagging:TagConcept'],
          thingSameAs: ['http://dbpedia.org/resource/Economy'],
          thingEnglishLabel: 'Economics',
          type: 'about',
        },
      ],
    },
  },
  id: 'urn:bbc:ares::article:crg7rvwrxdlo',
  type: 'optimo',
};
