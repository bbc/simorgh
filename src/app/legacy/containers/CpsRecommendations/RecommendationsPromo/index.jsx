import React, { useContext } from 'react';
import styled from '@emotion/styled';
import {
  GEL_SPACING,
  GEL_SPACING_DBL,
} from '#psammead/gel-foundations/src/spacings';
import {
  GEL_GROUP_1_SCREEN_WIDTH_MAX,
  GEL_GROUP_2_SCREEN_WIDTH_MIN,
  GEL_GROUP_3_SCREEN_WIDTH_MIN,
} from '#psammead/gel-foundations/src/breakpoints';
import { getSerifMedium } from '#psammead/psammead-styles/src/font-styles';
import { getPica } from '#psammead/gel-foundations/src/typography';
import { shape, string, oneOfType } from 'prop-types';
import { optimoStoryItem, storyItem } from '#models/propTypes/storyItem';

import { ARTICLE_PAGE } from '../../../../routes/utils/pageTypes';
import { RequestContext } from '../../../../contexts/RequestContext';
import { ServiceContext } from '../../../../contexts/ServiceContext';
import Grid from '../../../components/Grid';
import RecommendationsImage from '../RecommendationsPromoImage';
import useCombinedClickTrackerHandler from '../../StoryPromo/useCombinedClickTrackerHandler';
import extractPromoData from './utility';

const StyledPromoWrapper = styled.div`
  position: relative;
  padding: ${GEL_SPACING};
  margin-top: ${GEL_SPACING};
  background-color: ${props =>
    props.isArticlePage
      ? props.theme.palette.GHOST
      : props.theme.palette.GREY_2};
`;

const ImageWrapper = styled.div`
  display: inline-block;
  position: relative;
  width: 4.7rem;
  vertical-align: top;

  @media (min-width: ${GEL_GROUP_2_SCREEN_WIDTH_MIN}) {
    width: 6.8rem;
  }

  @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}) {
    width: 7.5rem;
  }
`;

const TextWrapper = styled.div`
  display: inline-block;
  width: calc(100% - 7.5rem);
  padding: 0 ${GEL_SPACING};
  vertical-align: top;
  height: 100%;

  @media (max-width: ${GEL_GROUP_1_SCREEN_WIDTH_MAX}) {
    width: calc(100% - 5rem);
  }

  @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}) {
    padding: 0 ${GEL_SPACING_DBL};
  }
`;

const Link = styled.a`
  position: static;
  color: ${props => props.theme.palette.GREY_10};
  text-decoration: none;
  overflow-wrap: break-word;

  &:before {
    bottom: 0;
    content: '';
    left: 0;
    overflow: hidden;
    position: absolute;
    right: 0;
    top: 0;
    white-space: nowrap;
    z-index: 1;
  }

  &:hover,
  &:focus {
    text-decoration: underline;
  }

  &:visited {
    color: ${props => props.theme.palette.METAL};
  }
`;

const StyledHeadline = styled.div`
  ${({ service }) => getSerifMedium(service)}
  ${({ script }) => getPica(script)}
  color: ${props => props.theme.palette.GREY_10};
  margin: 0;
  height: 100%;
  display: flex;
  align-items: center;
`;

const RecommendationsPromo = ({ promo, eventTrackingData }) => {
  const { script, service } = useContext(ServiceContext);
  const handleClickTracking = useCombinedClickTrackerHandler(eventTrackingData);

  const { headline, url, indexImage } = extractPromoData({ promo });

  const { pageType } = useContext(RequestContext);
  const isArticle = pageType === ARTICLE_PAGE;

  return (
    <Grid
      columns={{
        group0: 1,
        group1: 1,
        group2: 1,
        group3: 1,
        group4: 1,
        group5: 1,
      }}
      enableGelGutters
    >
      <StyledPromoWrapper
        data-e2e="story-promo-wrapper"
        isArticlePage={isArticle}
      >
        <ImageWrapper>
          <RecommendationsImage indexImage={indexImage} lazyLoad />
        </ImageWrapper>
        <TextWrapper>
          <StyledHeadline script={script} service={service}>
            <Link
              href={url}
              onClick={eventTrackingData ? handleClickTracking : null}
            >
              {headline}
            </Link>
          </StyledHeadline>
        </TextWrapper>
      </StyledPromoWrapper>
    </Grid>
  );
};

RecommendationsPromo.propTypes = {
  promo: oneOfType([shape(storyItem), shape(optimoStoryItem)]).isRequired,
  eventTrackingData: shape({
    block: shape({
      componentName: string,
    }),
    link: shape({
      componentName: string,
      url: string,
      format: string,
    }),
  }),
};

RecommendationsPromo.defaultProps = {
  eventTrackingData: null,
};

export default RecommendationsPromo;
