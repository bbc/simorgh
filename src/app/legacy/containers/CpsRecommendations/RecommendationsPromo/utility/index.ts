import path from 'ramda/src/path';
import hasPath from 'ramda/src/hasPath';
import { RecommendationsPromo } from '../../../../../models/types/recommendations';
import getOriginCode from '../../../../../lib/utilities/imageSrcHelpers/originCode';
import getLocator from '../../../../../lib/utilities/imageSrcHelpers/locator';

const extractPromoData = ({ promo }: { promo: RecommendationsPromo }) => {
  if (!promo) return null;

  if (hasPath(['indexImage'], promo)) {
    const cpsImage = path(['indexImage'], promo);

    return {
      headline: path(['headlines', 'headline'], promo),
      url: path(['locators', 'assetUri'], promo),
      indexImage: {
        width: path(['width'], cpsImage),
        height: path(['height'], cpsImage),
        altText: path(['altText'], cpsImage),
        copyrightHolder: path(['copyrightHolder'], cpsImage),
        originCode: getOriginCode(path(['path'], cpsImage)),
        locator: getLocator(path(['path'], cpsImage)),
      },
    };
  }

  const optimoImageBlocks = path<
    RecommendationsPromo['images']['defaultPromoImage']['blocks']
  >(['images', 'defaultPromoImage', 'blocks'], promo);

  const optimoImage = optimoImageBlocks?.find(
    block => block.type === 'rawImage',
  );
  const optimoImageAltText = optimoImageBlocks?.find(
    block => block.type === 'altText',
  );

  return {
    headline: path(
      [
        'headlines',
        'promoHeadline',
        'blocks',
        0,
        'model',
        'blocks',
        0,
        'model',
        'text',
      ],
      promo,
    ),
    url: path(['locators', 'canonicalUrl'], promo),
    indexImage: {
      width: path(['model', 'width'], optimoImage),
      height: path(['model', 'height'], optimoImage),
      altText: path(
        ['model', 'blocks', 0, 'model', 'blocks', 0, 'model', 'text'],
        optimoImageAltText,
      ),
      copyrightHolder: path(['model', 'copyrightHolder'], optimoImage),
      originCode: path(['model', 'originCode'], optimoImage),
      locator: path(['model', 'locator'], optimoImage),
    },
  };
};

export default extractPromoData;
