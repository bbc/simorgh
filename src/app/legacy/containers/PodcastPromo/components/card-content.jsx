import styled from '@emotion/styled';

import {
  GEL_SPACING,
  GEL_SPACING_DBL,
} from '#psammead/gel-foundations/src/spacings';
import {
  GEL_GROUP_2_SCREEN_WIDTH_MIN,
  GEL_GROUP_4_SCREEN_WIDTH_MIN,
} from '#psammead/gel-foundations/src/breakpoints';

const CardContent = styled.div`
  flex: 1;
  padding: ${GEL_SPACING};

  @media (min-width: ${GEL_GROUP_2_SCREEN_WIDTH_MIN}) {
    padding: ${GEL_SPACING} ${GEL_SPACING_DBL};
  }

  @media (min-width: ${GEL_GROUP_4_SCREEN_WIDTH_MIN}) {
    padding: ${GEL_SPACING_DBL};
  }
`;

export default CardContent;
