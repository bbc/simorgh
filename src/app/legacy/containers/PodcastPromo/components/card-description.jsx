import styled from '@emotion/styled';
import { getLongPrimer } from '#psammead/gel-foundations/src/typography';
import { getSansRegular } from '#psammead/psammead-styles/src/font-styles';
import { GEL_SPACING } from '#psammead/gel-foundations/src/spacings';

const CardDescription = styled.p`
  ${({ script }) => getLongPrimer(script)}
  ${({ service }) => getSansRegular(service)}
  max-width: 30rem;
  color: ${props => props.theme.palette.METAL};
  margin-top: ${GEL_SPACING};
`;

export default CardDescription;
