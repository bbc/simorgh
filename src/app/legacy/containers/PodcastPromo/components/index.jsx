import React from 'react';
import styled from '@emotion/styled';
import { string, shape, arrayOf, element } from 'prop-types';
import { scriptPropType } from '#psammead/gel-foundations/src/prop-types';
import { GEL_SPACING_DBL } from '#psammead/gel-foundations/src/spacings';

import Title from './title';
import Card from './card';
import CardContent from './card-content';
import CardLink from './card-link';
import CardImageWrapper from './card-image-wrapper';
import CardTitle from './card-title';
import CardDescription from './card-description';
import CardEpisodesText from './card-episodes-text';

const PodcastContext = React.createContext({});
const withPodcastContext = Component => props =>
  (
    <PodcastContext.Consumer>
      {context => <Component {...context} {...props} />}
    </PodcastContext.Consumer>
  );

const Wrapper = styled.section`
  background-color: ${props => props.theme.palette.LUNAR};
  padding: ${GEL_SPACING_DBL};
`;

const PodcastPromo = ({ script, service, children, ...props }) => (
  <PodcastContext.Provider value={{ script, service }}>
    <Wrapper {...props}>{children}</Wrapper>
  </PodcastContext.Provider>
);

PodcastPromo.Title = withPodcastContext(Title);
PodcastPromo.Card = Card;
PodcastPromo.Card.Link = CardLink;
PodcastPromo.Card.ImageWrapper = CardImageWrapper;
PodcastPromo.Card.Content = CardContent;
PodcastPromo.Card.Title = withPodcastContext(CardTitle);
PodcastPromo.Card.Description = withPodcastContext(CardDescription);
PodcastPromo.Card.EpisodesText = withPodcastContext(CardEpisodesText);

PodcastPromo.propTypes = {
  children: arrayOf(element).isRequired,
  script: shape(scriptPropType).isRequired,
  service: string.isRequired,
};

PodcastPromo.defaultProps = {};

export default PodcastPromo;
