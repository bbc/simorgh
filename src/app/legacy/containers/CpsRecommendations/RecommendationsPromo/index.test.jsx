/* eslint-disable react/prop-types */
import React from 'react';
import { ToggleContextProvider } from '#contexts/ToggleContext';
import { cpsRecommendation, optimoRecommendation } from './fixture';
import { render } from '../../../../components/react-testing-library-with-providers';
import { ServiceContextProvider } from '../../../../contexts/ServiceContext';
import RecommendationsPromo from '.';

const Component = ({ promo }) => {
  return (
    <ServiceContextProvider service="pidgin">
      <ToggleContextProvider
        toggles={{
          eventTracking: {
            enabled: true,
          },
        }}
      >
        <RecommendationsPromo promo={promo} dir="ltr" />,
      </ToggleContextProvider>
    </ServiceContextProvider>
  );
};

describe('RecommendationsPromo', () => {
  it('it renders a Story Promo wrapped in a Grid component', () => {
    const { container } = render(
      <RecommendationsPromo promo={cpsRecommendation} dir="ltr" />,
      {
        service: 'pidgin',
        toggles: {
          eventTracking: {
            enabled: true,
          },
        },
        pageType: 'STY',
      },
    );
    expect(container).toMatchSnapshot();
  });

  it('should render the title of the article as a link', () => {
    const { getByText, container } = render(
      <Component promo={cpsRecommendation} />,
    );

    const links = container.querySelectorAll('a');

    expect(
      getByText('Meet boys who dey convert cassava to electricity'),
    ).toBeInTheDocument();
    expect(links).toHaveLength(1);
    expect(links[0].getAttribute('href')).toEqual('/pidgin/44508901');
  });

  it('it should render recommendation correctly for optimo promos', () => {
    const { getByText, container } = render(
      <Component promo={optimoRecommendation} />,
    );

    const links = container.querySelectorAll('a');

    expect(
      getByText(
        'Merkez Bankası politika faizini neden indirdi, enflasyonu düşürmek için ne yapmalı?',
      ),
    ).toBeInTheDocument();
    expect(links).toHaveLength(1);
    expect(links[0].getAttribute('href')).toEqual(
      'https://www.bbc.com/turkce/articles/crg7rvwrxdlo',
    );
  });
});
