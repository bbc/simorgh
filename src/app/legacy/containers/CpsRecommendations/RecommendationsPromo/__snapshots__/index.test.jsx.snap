// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecommendationsPromo it renders a Story Promo wrapped in a Grid component 1`] = `
@supports (display: grid) {
  .emotion-0 {
    display: grid;
    position: initial;
    width: initial;
    margin: 0;
  }

  @media (max-width: 14.9375rem) {
    .emotion-0 {
      grid-template-columns: repeat(1, 1fr);
      grid-column-end: span 1;
      grid-column-gap: 0.5rem;
    }
  }

  @media (min-width: 15rem) and (max-width: 24.9375rem) {
    .emotion-0 {
      grid-template-columns: repeat(1, 1fr);
      grid-column-end: span 1;
      grid-column-gap: 0.5rem;
    }
  }

  @media (min-width: 25rem) and (max-width: 37.4375rem) {
    .emotion-0 {
      grid-template-columns: repeat(1, 1fr);
      grid-column-end: span 1;
      grid-column-gap: 0.5rem;
    }
  }

  @media (min-width: 37.5rem) and (max-width: 62.9375rem) {
    .emotion-0 {
      grid-template-columns: repeat(1, 1fr);
      grid-column-end: span 1;
      grid-column-gap: 1rem;
    }
  }

  @media (min-width: 63rem) and (max-width: 79.9375rem) {
    .emotion-0 {
      grid-template-columns: repeat(1, 1fr);
      grid-column-end: span 1;
      grid-column-gap: 1rem;
    }
  }

  @media (min-width: 80rem) {
    .emotion-0 {
      grid-template-columns: repeat(1, 1fr);
      grid-column-end: span 1;
      grid-column-gap: 1rem;
    }
  }
}

.emotion-2 {
  position: relative;
  padding: 0.5rem;
  margin-top: 0.5rem;
  background-color: #F6F6F6;
}

.emotion-4 {
  display: inline-block;
  position: relative;
  width: 4.7rem;
  vertical-align: top;
}

@media (min-width: 25rem) {
  .emotion-4 {
    width: 6.8rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-4 {
    width: 7.5rem;
  }
}

.emotion-6 {
  position: relative;
  height: 0;
  overflow: hidden;
  background-color: #F2F2F2;
  -webkit-background-position: center center;
  background-position: center center;
  background-repeat: no-repeat;
  -webkit-background-size: 60px 17px;
  background-size: 60px 17px;
  width: 100%;
  background-image: url(data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiPz48c3ZnIHdpZHRoPSI0NzkiIGhlaWdodD0iMTM2IiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciPjxnIGZpbGw9Im5vbmUiIGZpbGwtcnVsZT0iZXZlbm9kZCI+PGcgZmlsbD0iI0Q1RDBDRCI+PGc+PHBhdGggZD0iTTQ0OS41NTYgOTEuNTUzbC0yLjMxIDEuNDFjLTExLjE2NyA2LjgxOC0yMy4zMTMgMTAuNDc0LTM0LjM0NiAxMC40Ny0yMy42MS0uMDktMzkuMTYzLTE0LjA4My0zOS4yMjctMzQuNDUzLjAzLTE5LjkxOCAxNi4yNy0zNC42NjMgMzguNjMzLTM0LjcxOCAxMS4zODcuMDEgMjEuNzAzIDIuOTU0IDMzLjk2MiA5LjY3MmwyLjI1MSAxLjI0di0xOC4xOWwtLjk2Mi0uMzc3Yy0xMy44MjQtNS40NTgtMjQuNTMtNy44OS0zNS4xMDMtNy44ODQtMTYuMzQ2LS4wMDYtMzAuNTMzIDUuMzk0LTQwLjYzNyAxNC41NTctMTAuMTA1IDkuMTYzLTE2LjEwNiAyMi4xMDItMTYuMDk5IDM2Ljk1My4wMDggMTAuMzQ4IDQuMjc5IDIyLjQ4IDEzLjQyIDMyLjEwNSA5LjEyMSA5LjYyOCAyMy4xNjUgMTYuNjQ4IDQyLjQzIDE2LjYzOWguMDYzYzE1Ljk4IDAgMjcuMDYyLTMuNTYzIDM3LjA3NC04LjQ5MmwuODUxLS40MTRWOTEuNTUzek0zMzQgMTM1LjY5N2gxNDQuMTk1VjBIMzM0djEzNS42OTd6Ii8+PHBhdGggZD0iTTI3Ni45MzcgODkuOTY4Yy4wNDEtMTIuMzMtOC4xNzEtMjEuNjk2LTIxLjMwOC0yNS4zIDMuNTQ0LTEuODA5IDYuMzUtNC4wMjMgOC40MDQtNi43MjcgMi43NS0zLjYyMiA0LjA2MS04LjA2NCA0LjA0Ni0xMy4yMzUuMDE1LTYuMzU5LTIuNDg2LTEyLjgzOS03Ljg1OC0xNy42ODctNS4zNzItNC44NDctMTMuNTI2LTcuOTk3LTI0LjY1NC03Ljk5MUgyMDQuODN2OTcuNzI4aDM2LjA3M2MxMi44NyAwIDIxLjkwNi0zLjQ4MiAyNy43MjItOC42NSA1LjgxOC01LjE1NSA4LjMyLTExLjkxIDguMzEyLTE4LjEzOHpNMTY3IDEzNS42OThoMTQ0LjE5N1YwSDE2N3YxMzUuNjk3eiIvPjxwYXRoIGQ9Ik0xMDkuOTM3IDg5Ljk2OGMuMDQxLTEyLjMzLTguMTcxLTIxLjY5Ni0yMS4zMDgtMjUuMyAzLjU0NC0xLjgwOSA2LjM1LTQuMDIzIDguNDA0LTYuNzI3IDIuNzUtMy42MjIgNC4wNjEtOC4wNjQgNC4wNDYtMTMuMjM1LjAxNS02LjM1OS0yLjQ4Ni0xMi44MzktNy44NTgtMTcuNjg3LTUuMzcyLTQuODQ3LTEzLjUyNi03Ljk5Ny0yNC42NTQtNy45OTFIMzcuODN2OTcuNzI4aDM2LjA3M2MxMi44NyAwIDIxLjkwNi0zLjQ4MiAyNy43MjItOC42NSA1LjgxOC01LjE1NSA4LjMyLTExLjkxIDguMzEyLTE4LjEzOHpNMCAxMzUuNjk4aDE0NC4xOTdWMEgwdjEzNS42OTd6Ii8+PHBhdGggZD0iTTI1OC42NjIgODguMTk4Yy0uMDEzIDMuMjI5LTEuMDA3IDYuNDc1LTMuODk2IDkuMDExLTIuODg0IDIuNTM3LTcuODczIDQuNDYzLTE2LjEzMyA0LjQ2M0gyMjJWNzVoMTUuODkzYzcuNDExIDAgMTIuNjcgMS41MDIgMTUuOTY1IDMuODUgMy4yODkgMi4zNjIgNC43NzYgNS40NjMgNC44MDQgOS4zNDgiLz48cGF0aCBkPSJNOTEuNjYyIDg4LjE5OGMtLjAxMyAzLjIyOS0xLjAwNyA2LjQ3NS0zLjg5NiA5LjAxMS0yLjg4NCAyLjUzNy03Ljg3NCA0LjQ2My0xNi4xMzMgNC40NjNINTVWNzVoMTUuODkyYzcuNDEyIDAgMTIuNjcyIDEuNTAyIDE1Ljk2NiAzLjg1IDMuMjg5IDIuMzYyIDQuNzc2IDUuNDYzIDQuODA0IDkuMzQ4Ii8+PHBhdGggZD0iTTI0NS4xODYgNTUuNzljMy4wOTYtMi4yMzcgNC41OS01LjM4NiA0LjYxMy0xMC4xMjQtLjAxNS0zLjI1LS45NDMtNi4wMzMtMy4yODEtOC4xMTEtMi4zNDYtMi4wNzgtNi4zMy0zLjU1NS0xMi43NTQtMy41NTVIMjIydjI1LjI3NWg4LjA3NmM2Ljk4OC4wMDQgMTEuOTk4LTEuMjQzIDE1LjExLTMuNDg2Ii8+PHBhdGggZD0iTTc4LjE4NiA1NS43OWMzLjA5Ni0yLjIzNyA0LjU5LTUuMzg2IDQuNjEzLTEwLjEyNC0uMDE1LTMuMjUtLjk0My02LjAzMy0zLjI4Mi04LjExMUM3Ny4xNzIgMzUuNDc3IDczLjE4OCAzNCA2Ni43NjQgMzRINTV2MjUuMjc1aDguMDc2YzYuOTg4LjAwNCAxMS45OTgtMS4yNDMgMTUuMTEtMy40ODYiLz48L2c+PC9nPjwvZz48L3N2Zz4K);
}

@media (min-width: 25rem) {
  .emotion-6 {
    -webkit-background-size: 77px 22px;
    background-size: 77px 22px;
  }
}

@media (min-width: 63rem) {
  .emotion-6 {
    -webkit-background-size: 93px 27px;
    background-size: 93px 27px;
  }
}

.emotion-8 {
  display: inline-block;
  width: calc(100% - 7.5rem);
  padding: 0 0.5rem;
  vertical-align: top;
  height: 100%;
}

@media (max-width: 24.9375rem) {
  .emotion-8 {
    width: calc(100% - 5rem);
  }
}

@media (min-width: 37.5rem) {
  .emotion-8 {
    padding: 0 1rem;
  }
}

.emotion-10 {
  font-family: Helmet,Freesans,Helvetica,Arial,sans-serif;
  font-weight: 700;
  font-style: normal;
  font-size: 0.9375rem;
  line-height: 1.25rem;
  color: #141414;
  margin: 0;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media (min-width: 20rem) and (max-width: 37.4375rem) {
  .emotion-10 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-10 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

.emotion-12 {
  position: static;
  color: #141414;
  -webkit-text-decoration: none;
  text-decoration: none;
  overflow-wrap: break-word;
}

.emotion-12:before {
  bottom: 0;
  content: '';
  left: 0;
  overflow: hidden;
  position: absolute;
  right: 0;
  top: 0;
  white-space: nowrap;
  z-index: 1;
}

.emotion-12:hover,
.emotion-12:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-12:visited {
  color: #6E6E73;
}

<div>
  <div
    class="emotion-0 emotion-1"
    dir="ltr"
  >
    <div
      class="emotion-2 emotion-3"
      data-e2e="story-promo-wrapper"
    >
      <div
        class="emotion-4 emotion-5"
      >
        <div
          class="emotion-6 emotion-7"
          data-e2e="image-placeholder"
          style="padding-bottom: 56.25%;"
        >
          <div
            class="lazyload-wrapper "
          >
            <div
              class="lazyload-placeholder"
            />
          </div>
        </div>
      </div>
      <div
        class="emotion-8 emotion-9"
      >
        <div
          class="emotion-10 emotion-11"
        >
          <a
            class="emotion-12 emotion-13"
            href="/pidgin/44508901"
          >
            Meet boys who dey convert cassava to electricity
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
